multiprofit_task.py

```diff
@@ -56,9 +87,12 @@ class MultiProFitConfig(pexConfig.Config):
     estimateContiguousDenoisedMoments = pexConfig.Field(
         dtype=bool, default=True, doc="Whether models initiated from moments should estimate within "
                                       "contiguous positive pixels in a naively de-noised image")
-    filenameOut = pexConfig.Field(dtype=str, default="", doc="Filename for output of FITS table")
-    filenameOutDeblend = pexConfig.Field(dtype=str, default="", doc="Filename for output of FITS table with"
-                                                                    " deblended fits")
+    field_localbg = pexConfig.Field(
+        dtype=str, default='base_LocalBackground', doc="Field name (prefix to _instFlux) to read the local background "
+                                                       "level from if usePriorBackgroundLocalEstimate is True")
+    filenameOut = pexConfig.Field(dtype=str, optional=True, doc="Filename for output of FITS table")
+    filenameOutDeblend = pexConfig.Field(dtype=str, optional=True, doc="Filename for output of FITS table with"
+                                                                      " deblended fits")
```

Comment:
The code that uses these options should never be run inside a `PipelineTask`, because all I/O needs to go through butler.  If it's hard to factor them out of the subtask that the `PipelineTask` will use, perhaps the base subtask interface should define a method like
```
@classmethod
def validateConfigsForPipelineTask(cls, config: ConfigClass) -> None:
    """Raise if the given config includes settings that are illegal when used in a PipelineTask.
    """.
    pass  # assume most subclasses don't have any such options
```
This task could then implement that to check that `config.filenameOut is None` (etc), and the parent `PipelineTask` could call it.

Reply:
I intend to refactor the deblending-from-previous-single-source-fitting-catalogs into a separate `PipelineTask` in another ticket; it's too substantial of a change to include here.